modules/RTC/RTCUtils.js

```diff
@@ -420,26 +423,14 @@ function newGetConstraints(um = [], options = {}) {
                 noiseSuppression: !disableNS && !disableAP
             };
         } else {
-            // NOTE(brian): the new-style ('advanced' instead of 'optional')
-            // doesn't seem to carry through the googXXX constraints
-            // Changing back to 'optional' here (even with video using
-            // the 'advanced' style) allows them to be passed through
-            // but also requires the device id to capture to be set in optional
-            // as sourceId otherwise the constraints are considered malformed.
-            if (!constraints.audio.optional) {
-                constraints.audio.optional = [];
-            }
-            constraints.audio.optional.push(
-                { sourceId: options.micDeviceId },
-                { echoCancellation: !disableAEC && !disableAP },
-                { googEchoCancellation: !disableAEC && !disableAP },
-                { googAutoGainControl: !disableAGC && !disableAP },
-                { googNoiseSuppression: !disableNS && !disableAP },
-                { googHighpassFilter: !disableHPF && !disableAP },
-                { googNoiseSuppression2: !disableNS && !disableAP },
-                { googEchoCancellation2: !disableAEC && !disableAP },
-                { googAutoGainControl2: !disableAGC && !disableAP }
-            );
+            constraints.audio = {
+                autoGainControl: { exact: !disableAGC && !disableAP },
+                channelCount,
```

Comment:
Maybe this should also be exact, what do you guys think?

Reply:
I think "ideal" would best. We still want high quality audio even if you have a shitty microphone. We want to allow you to *receive* it too.

Reply:
I agree, we should request `ideal` rather than `exact`. Exact would make it a required setting and it would result in a gUM error if it can't be met. Also, if we are getting rid of the old-style of constraints for ChromiumBased clients, why not remove the browser check above and do the same across all browsers ?